File: dataminer/MinerWrapper/GpuInfo.py
from CommonKeys import *
from MinerKeys import *
import ServerInfo


ServerInfo = ServerInfo.ServerInfo()

class GpuInfo():


    def processDevs(self, devs, time):
        devs_json = []
        for dev in devs:
            devs_json.append(self.processDev(dev, time))
        return devs_json

    def processDev(self, dev, time):
        dev_json = {}
        dev_json[CommonKeys.GPU_ID] = ServerInfo.getGPUGlobalId(dev[MinerKeys.GPU_ID])
        dev_json[CommonKeys.TEMPERATURE] = dev[MinerKeys.TEMPERATURE]
        dev_json[CommonKeys.GPU_VOLTAGE] = dev[MinerKeys.GPU_VOLTAGE]
        dev_json[CommonKeys.GPU_CLOCK] = dev[MinerKeys.GPU_CLOCK]
        dev_json[CommonKeys.MEM_CLOCK] = dev[MinerKeys.MEM_CLOCK]
        dev_json[CommonKeys.FAN_RPM] = dev[MinerKeys.FAN_RPM]
        dev_json[CommonKeys.HW_ERRORS] = dev[MinerKeys.HW_ERRORS]
        dev_json[CommonKeys.REJECTED] = dev[MinerKeys.REJECTED]
        dev_json[CommonKeys.ACCEPTED] = dev[MinerKeys.ACCEPTED]
        dev_json[CommonKeys.CURRENT_HASH_RATE] = int(dev[MinerKeys.CURRENT_HASH_RATE] * 1000)
        dev_json[CommonKeys.INTENSITY] = int(dev[MinerKeys.INTENSITY])
        dev_json[CommonKeys.TIME_SINCE_LAST_WORK] = time - dev[MinerKeys.LAST_WORK_TIME]
        dev_json[CommonKeys.TIME_SINCE_LAST_VALID_WORK] = time - dev[MinerKeys.LAST_VALID_WORK_TIME]
        return dev_json

File: AnalyticsJava/src/analytics/data/GpuInfo.java
package analytics.data;

import org.json.simple.JSONObject;

import analytics.Dal;
import analytics.constants.Errors;
import analytics.constants.Keys;

public class GpuInfo {

	public int cardId;
	public int timestamp;
	public float temperature;
	public float voltage;
	public int coreClock;
	public int memClock;
	public int fanRpm;
	public int hwErrors;
	public int sharesAccepted;
	public int sharesRefused;
	public int currentHashRate;
	public int intensity;
	public int timeSinceLastWork;
	public int timeSinceLastValidWork;

	public int validSharedSinceLast;
	public int invalidSharedSinceLast;

	public GpuInfo() {

	}

	public GpuInfo(JSONObject object, int timestamp) {

		this.timestamp = timestamp;

		try {
			this.cardId = ((Long) object.get(Keys.GPU_ID)).intValue();
			this.temperature = ((Double) object.get(Keys.TEMPERATURE)).floatValue();
			this.voltage = ((Double) object.get(Keys.GPU_VOLTAGE)).floatValue();
			this.coreClock = ((Long) object.get(Keys.GPU_CLOCK)).intValue();
			this.memClock = ((Long) object.get(Keys.MEM_CLOCK)).intValue();
			this.fanRpm = ((Long) object.get(Keys.FAN_RPM)).intValue();
			this.hwErrors = ((Long) object.get(Keys.HW_ERRORS)).intValue();
			this.sharesAccepted = ((Long) object.get(Keys.ACCEPTED)).intValue();
			this.sharesRefused = ((Long) object.get(Keys.REJECTED)).intValue();
			this.currentHashRate = ((Long) object.get(Keys.CURRENT_HASH_RATE)).intValue();
			this.intensity = ((Long) object.get(Keys.INTENSITY)).intValue();
			this.timeSinceLastWork = ((Long) object.get(Keys.TIME_SINCE_LAST_WORK)).intValue();
			this.timeSinceLastValidWork = ((Long) object.get(Keys.TIME_SINCE_LAST_VALID_WORK)).intValue();

		} catch (NullPointerException e) {
			Dal.log(Errors.RESPONSE_MISSING_KEY, "Bad object received from miner wrapper");
		}

	}
}
